src/network/tc/qdisc.c

```diff
@@ -250,15 +254,25 @@ int config_parse_tc_qdiscs_parent(
 
         if (streq(rvalue, "root"))
                 qdisc->parent = TC_H_ROOT;
-        else if (streq(rvalue, "clsact"))
+        else if (streq(rvalue, "clsact")) {
                 qdisc->parent = TC_H_CLSACT;
-        else {
+                qdisc->handle = TC_H_MAKE(TC_H_CLSACT, 0);
+        } else if (streq(rvalue, "ingress")) {
+                qdisc->parent = TC_H_INGRESS;
+                qdisc->handle = TC_H_MAKE(TC_H_INGRESS, 0);
+        } else {
                 log_syntax(unit, LOG_ERR, filename, line, r,
                            "Failed to parse [QueueDiscs] 'Parent=', ignoring assignment: %s",
                            rvalue);
                 return 0;
         }
 
+        if (!streq(rvalue, "root")) {
+                qdisc->tca_kind = strdup(rvalue);
```

Comment:
No no, please use `free_and_strdup()`, and `free()` this when `rvalue` is `"root"`. Otherwise,
```
Parent = clsact
Parent = root
```
or
```
Parent = clsact
Parent = ingress
```
causes memleak.